src/finch/finch_logic/nodes.py

```diff
@@ -488,23 +432,21 @@ class Query(LogicNode):
     lhs: LogicNode
     rhs: LogicNode
 
-    @staticmethod
-    def is_expr():
-        """Determines if the node is an expression."""
-        return True
-
     @staticmethod
     def is_stateful():
         """Determines if the node is stateful."""
         return True
 
-    def children(self):
+    def children(self) -> list[LogicNode]:
         """Returns the children of the node."""
         return [self.lhs, self.rhs]
 
+    def get_fields(self) -> list[Field]:
```

Comment:
@mtsokol I'm not entirely happy about this one (and the ones below). Can this be implemented?

Reply:
I think it can (`Subquery` has it) but let's leave it for now, and decide in the call (at the end it isn't used anywhere probably).

Reply:
If it can be implemented, then it's fine to leave it as `raise NotImplementedError` for now.